### PR TITLE
additional completion providers

### PR DESCRIPTION
**Pull request by havvg**
- The indentation of all files has been synchronised.
- The multi-purpose files have been separated into multiple files.

### CODE REVIEW

**Comment by havvg** [comment]
I hope the files are not required to be named `.ide...` because that's pretty annoying within this repository, as those files are invisible by convention, which makes sense on a project scale, but not here, imho.

File: doctrine/common-toolbox.metadata.json [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+            "signature": [
+                "Doctrine\\Common\\Persistence\\ManagerRegistry::getManagerForClass",
+                "Doctrine\\Common\\Persistence\\ObjectManager::find",
+                "Doctrine\\Common\\Persistence\\ObjectManager::getClassMetadata",
+                "Doctrine\\Common\\Persistence\\ObjectManager::getRepository"
+            ],
+            "signatures": [
+                {
+                    "class": "Doctrine\\Common\\Persistence\\ManagerRegistry",
+                    "method": "getManagerForClass",
```
Comment (Koc) [inline doctrine/common-toolbox.metadata.json]: Get manager for class returns object manager interface, not entity class

File: doctrine/common-toolbox.metadata.json [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+                    "type": "type"
+                }, {
+                    "class": "Doctrine\\Common\\Persistence\\ObjectManager",
+                    "method": "find",
+                    "type": "type"
+                }, {
+                    "class": "Doctrine\\Common\\Persistence\\ObjectManager",
```
Comment (Koc) [inline doctrine/common-toolbox.metadata.json]: Please add also getReference method

File: doctrine/common-toolbox.metadata.json [hunk 3]
```diff
@@ -0,0 +1,29 @@
+{
+    "registrar": [
+        {
+            "language": "php",
```
Comment (Koc) [inline doctrine/common-toolbox.metadata.json]: I think that we should use different provider here, like 'doctrine.entities' but it not supported now, //cc @Haehnchen

File: doctrine/common-toolbox.metadata.json (hunk 2 above)
Comment (Koc) [inline doctrine/common-toolbox.metadata.json]: See https://github.com/Haehnchen/idea-php-symfony2-plugin/issues/441

File: doctrine/common-toolbox.metadata.json (hunk 3 above)
Comment (Haehnchen) [inline doctrine/common-toolbox.metadata.json]: added `doctrine.models` as also documents are supported

**Comment by havvg** [comment]
Please don't merge yet. I can't verify it's working at: `PHPStorm 10.0.1 with PHP Toolbox 0.2.2` (PHPStorm 11 EAP, too). It seems the new filenames are not read correctly. When I switch back to the master branch of my repository, it's fine.

@Haehnchen can you verify the directory structure is valid? It looks like the plugin doesn't find the new files. When I move the files into root directory, everything is fine, but named files in sub-directory are not read.

**Comment by Koc** [comment]
@havvg see https://github.com/Haehnchen/idea-php-toolbox/issues/12

File: doctrine/common-toolbox.metadata.json [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+            "provider": "doctrine.models",
+            "signature": [
+                "Doctrine\\Common\\Persistence\\ManagerRegistry::getManagerForClass",
+                "Doctrine\\Common\\Persistence\\ObjectManager::find",
```
Comment (Koc) [inline doctrine/common-toolbox.metadata.json]: same signature described below in more extended format. So this line is duplicates

File: doctrine/orm-toolbox.metadata.json [hunk 5]
```diff
@@ -0,0 +1,30 @@
+{
+    "registrar": [
+        {
+            "language": "php",
+            "provider": "doctrine.models",
+            "signature": [
+                "Doctrine\\ORM\\EntityManagerInterface::copy",
```
Comment (Koc) [inline doctrine/orm-toolbox.metadata.json]: Why you are duplicate this methods in short notation? It is already described in `signatures` section

Comment (havvg) [inline doctrine/orm-toolbox.metadata.json]: That's not the same.

The entry in `signature` responds to: `$reference = $entityManager->getReference('<CARET>', 5);`; providing autocompletion for all `doctrine.models`.
Afterwards, the entry in `signatures` responds to `$reference-><CARET>`; providing autocompletion on `$reference` based on the type (index 1 by default) into `getReference` call.

But you are right, the `copy` makes no sense here, as you pass in the object to copy, not a string.

**Comment by havvg** [comment]
@Koc @Haehnchen I read the discussion on Haehnchen/idea-php-toolbox#12, but I don't get it. When I move all the files of this pull request into the root directory of this repository leaving the names as they currently are, it works. It just doesn't work for sub folders. Did something change since then?

File: doctrine/common-toolbox.metadata.json (hunk 4 above)
Comment (havvg) [inline doctrine/common-toolbox.metadata.json]: That's not the same.

The entry in `signature` responds to: `$reference = $entityManager->getReference('<CARET>', 5);`; providing autocompletion for all `doctrine.models`.
Afterwards, the entry in `signatures` responds to `$reference-><CARET>`; providing autocompletion on `$reference` based on the type (index 1 by default) into `getReference` call.

But you are right, the `copy` makes no sense here, as you pass in the object to copy, not a string.

**Comment by Haehnchen** [comment]
please name them `.ide-toolbox.metadata.json`. i use a simple submodule to maintain them. that are only valid names in project level.

for a plugin build the namings are irrelevant.

File: doctrine/common-toolbox.metadata.json (hunk 4 above)
Comment (Haehnchen) [inline doctrine/common-toolbox.metadata.json]: what's about symfony plugin completion, does this results in duplicate entries in completion popover?

Comment (havvg) [inline doctrine/common-toolbox.metadata.json]: Not for me, latest versions:

screen shot 2016-02-03 at 20 53 39
screen shot 2016-02-03 at 20 54 00

**Comment by havvg** [comment]
@Haehnchen Good to go!

**Comment by Haehnchen** [comment]
thx.
deployment on weekend; will also normalize provider names

**Pull request merged**